agilerl/wrappers/pettingzoo_wrappers.py

```diff
@@ -0,0 +1,119 @@
+from __future__ import annotations
+
+import warnings
+
+import gymnasium.spaces
+import numpy as np
+from gymnasium.utils import seeding
+from pettingzoo.utils.env import ActionType, AgentID, ObsType, ParallelEnv
+
+
+class PettingZooBaseParallelWrapper(ParallelEnv[AgentID, ObsType, ActionType]):
```

Comment:
I have not implemented the vectorization in the wrapper because it is a more involved process for multiagent envs, I had to make some adjustments over the existing implementations. Either case, I believe AutoResetWrapper can exist independently.  I am providing a link to a discussion related to this as well as my vectorization method:

[https://github.com/Farama-Foundation/SuperSuit/issues/43#issuecomment-734461360](https://github.com/Farama-Foundation/SuperSuit/issues/43#issuecomment-734461360)

Reply:
Thanks for the link. I think vectorized envs would be really useful, and would need autoresetting, but agree that autoresetting is still useful on its own. For vectorised envs, do you have your own implementation which is more generalised to any environment?

Reply:
I don’t think my implementation would be generalizable either. I need to work on that separately and write an extensive testing routine if we want it to be a general implementation.

The fact that farama devs deliberately decided not to implement it does not give me confidence though…